Core/GDCore/IDE/Events/ExpressionCompletionFinder.h

```diff
@@ -209,31 +269,99 @@ class GD_CORE_API ExpressionCompletionFinder
     completions.push_back(
-        ExpressionCompletionDescription::ForObject(node.type, ""));
+        ExpressionCompletionDescription::ForObject(
+            node.type,
+            "",
+            searchedPosition + 1,
+            searchedPosition + 1));
     completions.push_back(
-        ExpressionCompletionDescription::ForExpression(node.type, ""));
+        ExpressionCompletionDescription::ForExpression(
+            node.type,
+            "", 
+            searchedPosition + 1,
+            searchedPosition + 1));
   }
   void OnVisitOperatorNode(OperatorNode& node) override {
     completions.push_back(
-        ExpressionCompletionDescription::ForObject(node.type, ""));
+        ExpressionCompletionDescription::ForObject(
+            node.type,
+            "",
+            searchedPosition + 1,
+            searchedPosition + 1));
     completions.push_back(
-        ExpressionCompletionDescription::ForExpression(node.type, ""));
+        ExpressionCompletionDescription::ForExpression(
+            node.type,
+            "",
+            searchedPosition + 1,
+            searchedPosition + 1));
   }
   void OnVisitUnaryOperatorNode(UnaryOperatorNode& node) override {
     completions.push_back(
-        ExpressionCompletionDescription::ForObject(node.type, ""));
+        ExpressionCompletionDescription::ForObject(
+            node.type,
+            "",
+            searchedPosition + 1,
+            searchedPosition + 1));
     completions.push_back(
-        ExpressionCompletionDescription::ForExpression(node.type, ""));
+        ExpressionCompletionDescription::ForExpression(
+            node.type,
+            "",
+            searchedPosition + 1,
+            searchedPosition + 1));
   }
   void OnVisitNumberNode(NumberNode& node) override {
     // No completions
   }
   void OnVisitTextNode(TextNode& node) override {
-    // No completions
+    FunctionCallNode* functionCall = dynamic_cast<FunctionCallNode*>(parentNodeAtLocation);
+    if (functionCall != NULL) {
+      int parameterIndex = -1;
+      for (int i = 0; i < functionCall->parameters.size(); i++)
+      {
+        if (functionCall->parameters.at(i).get() == &node) {
+          parameterIndex = i;
+          break;
+        }
+      }
+      if (parameterIndex < 0) {
+        return;
+      }
+      // Search the metadata parameter index skiping invisible ones
+      int visibleParameterIndex = 0;
+      int metadataParameterIndex = 0;
+      while (metadataParameterIndex < functionCall->expressionMetadata.parameters.size()) {
+        if (!functionCall->expressionMetadata.parameters[metadataParameterIndex].IsCodeOnly()) {
+          if (visibleParameterIndex == parameterIndex) {
+            break;
+          }
+          visibleParameterIndex++;
+        }
+        metadataParameterIndex++;
+      }
+      const gd::String &type = functionCall->expressionMetadata.parameters[metadataParameterIndex].GetType();
```

Comment:
Will this crash if there is for example no parameter in the expression metadata but you still tried to write a parameter in the function?

Reply:
I think it only worked because no completion is asked by the GUI when there is a parsing error.